pyiron_base/database/tables.py

```diff
@@ -0,0 +1,52 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+"""
+Classes defining the shape of pyiron's database tables.
+"""
+
+from sqlalchemy import (
+    Column,
+    DateTime,
+    Float,
+    Integer,
+    String,
+    Table,
+)
+
+__author__ = "Murat Han Celik, Liam Huber"
+__copyright__ = (
+    "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH"
+    " - Computational Materials Design (CM) Department"
+)
+__version__ = "0.0"
+__maintainer__ = "Liam Huber"
+__email__ = "huber@mpie.de"
+__status__ = "development"
+__date__ = "Sep, 2021"
+
+
+class HistoricalTable(Table):
+    """The historical table."""
+    def _init(self, table_name, metadata, *args, extend_existing=True, **kwargs):
+        super()._init(
+            table_name,
+            metadata,
+            Column("id", Integer, primary_key=True, autoincrement=True),
+            Column("parentid", Integer),
+            Column("masterid", Integer),
+            Column("projectpath", String(50)),
+            Column("project", String(255)),
+            Column("job", String(50)),
+            Column("subjob", String(255)),
+            Column("chemicalformula", String(30)),
```

Comment:
The limit for chemicalformula and hamilton has been increased to 50. In `generic.py`, it is set by `self._chem_formula_lim_length = 50`.

Reply:
Good catch! I was just recklessly copying from the old or

Reply:
I think it can be a good idea to have a limit for each column. Then if necessary we can change them once somewhere, and it is reflected throughout the code.

Reply:
Yeah, I 100% agree. I have an idea for a more ambitious refactoring that should accomplish this. Will try to get it up before the tech dev meeting